src/node.cc

```diff
@@ -981,6 +983,42 @@ void SetupNextTick(const FunctionCallbackInfo<Value>& args) {
       FIXED_ONE_BYTE_STRING(args.GetIsolate(), "_setupNextTick"));
 }
 
+void PromiseRejectCallback(PromiseRejectMessage message) {
+  Isolate* isolate = Isolate::GetCurrent();
```

Comment:
Since this is done inside `PromiseRejectCallback` which is passed to the isolate via `isolate->SetPromiseRejectCallback` it can likely be passed to the function in some way - the caller site has access to the isolate and the environment. I would suggest how but I'm not sure since I'm not familiar enough with the code base but it shouldn't be too hard.